ADR/00xx-component-references.md

```diff
@@ -0,0 +1,145 @@
+  * A new set of functionality for [build-service] uses the `ComponentReference` links to propose updates to users git repositories.
+* Introduce a new convention for PRs that lets one PR declare that it depends on another PR.
+* Important to understand: the declared reference embeddings as defined in the `ComponentReferences` and PR dependencies are two different things.
+  * `ComponentReferences` are declarations about which Components we should expect to see PR dependencies for.
+    * You say in english: “Component A references Component B.”
+  * PR dependencies can be supplied by the user without any declared `ComponentReferences` on the respective Components.
+    * You say in english: “PR #1 depends on PR #2.”
+
+### Integration-service
+
+* When [integration-service] notices a build of a *referent* Component (where another Component declares that it embeds a reference to this one), it always skips all testing (both pre-merge and post-merge). It always promotes the image to the global candidate list when the PR is merged, but it does not promote to dev environment (post-merge) and it does not create Releases (post-merge).
+  * Builds of *referent* Components never directly trigger tests, never trigger promotion, or Releases.
+* When [integration-service] notices a build of a *referer* Component (one that declares that it embeds references to others), it does testing as normal, promotes to global candidate list as normal, and it promotes to dev env as normal, and it creates Releases as normal.
+* When [integration-service] notices a build of a Component that references no other components and no other components refer to it - then it proceeds like normal, runs tests, promotes, deploys, releases, etc.
+
+### Build-service
+
+* [build-service] will propagate the digest from one Component to another as a PR – following the declared `ComponentReferences`. This is a bit like renovatebot wrapped in a controller. The [build-service] acts when it sees that a build pipelinerun has completed successfully for a *referent* image.
+  * If this is the first time it has seen this *referent* updated in this way, it will file a new pull request on the *referer* supplying the pullspec and digest of the new referent, using the same branch name as was used for the *referent* PR.
+  * If a PR on that branch already exists, update that branch with an additional commit including the new *referent* digest.
+
+The [build-service] will also update the PR it filed when the PRs that triggered it are merged or updated. It may update the description and/or it may rebase on the main branch and/or it may issue /retest. Need to figure out what we want here.
+
+### “PR dependency”
+
+TODO - describe exactly how the user specifies that one PR depends on another. PaC will have to pass this through to the PipelineRun as an annotation. For now, just assume it is possible and defined.
```

Comment:
TODO

Reply:
One idea:

* Indicate this with some special syntax in PR description: `Depends on: other/repo#123`
* Indicate this with some special comment with a syntax that PaC recognizes: `/depends-on other/repo#123`
* Indicate this by sharing branch names. If the two PRs share the same origin branch name (and same fork owner/organization) then they should be understood to be related. But, which one depends on which?

Reply:
Do all of these options depend on changes to PaC? If we are not proposing the addition of any dependency abstraction to PaC, then I am not sure we will be able to leverage it for making these links.

Reply:
Ack - yes, I think they all depend on PaC passing some new information along on the PipelineRuns it creates.

(I guess, the origin branch name could already be present - passed along by PaC. The other two would surely be new.)

Reply:
Today, I pushed 63dbb7c, updating this to use a *prefix* of the **source branch name** as a "PR Group" identifier.

I think this can work and it will require only low-complexity changes in PaC (all that needs to change is to start passing along the source branch name as an annotation to the PipelineRun).